src/lib/Runner.ts

```diff
@@ -223,32 +223,34 @@ export class Runner {
         this.config.mergeSettings &&
         this.config.mergeSettings.skipBuildOnDependentsAwaitingMerge;
       const pullRequestId = landRequest.pullRequestId;
-      Logger.verbose('Attempting merge pull request', {
+      Logger.info('Triggering merge attempt', {
         namespace: 'lib:runner:moveFromAwaitingMerge',
         pullRequestId,
         landRequestId: landRequest.id,
         lockId,
       });
-      await this.client.mergePullRequest(landRequestStatus, { skipCI });
-      Logger.info('Successfully merged PR', {
-        namespace: 'lib:runner:moveFromAwaitingMerge',
-        landRequestId: landRequest.id,
-        pullRequestId,
-        lockId,
-      });
 
-      const end = Date.now();
-      const queuedDate = await this.getLandRequestQueuedDate(landRequest.id);
-      const start = queuedDate!.getTime();
-      eventEmitter.emit('PULL_REQUEST.MERGE.SUCCESS', {
-        landRequestId: landRequestStatus.requestId,
-        pullRequestId: landRequest.pullRequestId,
-        commit: landRequest.forCommit,
-        sourceBranch: pullRequest.sourceBranch,
-        targetBranch: pullRequest.targetBranch,
-        duration: end - start,
-      });
-      return landRequest.setStatus('success');
+      await landRequest.setStatus('merging');
+      this.client.mergePullRequest(
+        landRequestStatus,
+        async () => {
```

Comment:
Is there any difference between using a callback here vs using `Promise.then`? Might be simpler to use `Promise.then`.

Reply:
Yeah probably cleaner to rework the `try...catch` into using `.then().catch()` so we can also make sure that errors are caught.